exec/Miner.hs

```diff
@@ -237,15 +239,20 @@ mining go = do
 -- at an higher rate.
 --
 miningLoop :: UpdateMap -> (TargetBytes -> HeaderBytes -> RIO Env HeaderBytes) -> RIO Env ()
-miningLoop updateMap inner = mask $ \umask ->
-    void (go umask) `finally` logInfo "Mining halted."
+miningLoop updateMap inner = mask go
   where
-    go :: (RIO Env () -> RIO Env a) -> RIO Env b
-    go umask = do
-        forever (umask loopBody) `catchAny` \e -> do
-            logWarn "Error in mining loop. Trying again ..."
-            logDebug $ display e
-        go umask
+    go :: (RIO Env () -> RIO Env a) -> RIO Env ()
+    go umask = (forever (umask loopBody) `catches` handlers) >>= \case
+        Irrecoverable -> pure ()
+        Recoverable   -> go umask
```

Comment:
What about adding a short delay here, say 1 second, to keep the miner spinning from too hot?